src/test/scala/org/bitcoins/core/protocol/blockchain/BlockHeaderSpec.scala

```diff
@@ -0,0 +1,14 @@
+package org.bitcoins.core.protocol.blockchain
+
+import org.bitcoins.core.gen.BlockchainElementsGenerator
+import org.scalacheck.{Prop, Properties}
+
+/**
+  * Created by tom on 7/6/16.
+  */
+class BlockHeaderSpec extends Properties("BlockHeaderSpec") {
+  property("serialization symmetry") =
+    Prop.forAll(BlockchainElementsGenerator.blockHeader) { header =>
+      BlockHeader(header.version, header.previousBlockHash, header.merkleRootHash, header.time, header.nBits, header.nonce) == header
```

Comment:
This doesn't test serialization symmetry, you need to serialize it to it's 'hex' or 'byte' representation, then use one of the 'apply' functions inside of BlockHeader` to read that serialization.